src/Application.vala

```diff
@@ -29,23 +29,28 @@ namespace Switchboard {
         public Gtk.SearchEntry search_box { get; private set; }
 
         private string all_settings_label = _("All Settings");
-        private uint configure_id;
 
         private Gee.ArrayList <Switchboard.Plug> previous_plugs;
         private Gee.LinkedList <string> loaded_plugs;
         private Gtk.Button navigation_button;
         private Gtk.HeaderBar headerbar;
         private Gtk.Stack stack;
-        private Hdy.Window main_window;
+        private MainWindow main_window;
         private Switchboard.CategoryView category_view;
         private Switchboard.Plug current_plug;
 
+        public static GLib.Settings settings;
+
         private static bool opened_directly = false;
         private static bool should_animate_next_transition = true;
         private static string? link = null;
         private static string? open_window = null;
         private static string? plug_to_open = null;
 
+        static construct {
+            settings = new GLib.Settings ("io.elementary.switchboard.saved-state");
```

Comment:
I'm unsure it is necessary to have it static. We are always sure to have just one instance of Switchboard.Application

Reply:
Is there a downside?  I presume this was done because it is slightly less code to access in MainWindow when static.
`SwitchboardApp.settings` compared to `(SwitchboardApp)application.settings` although that could be reduced by having a variable set to `(SwitchboardApp)application`.

Reply:
It makes the memory allocation easier to understand and ensures that we are not keeping a reference of this object when we don't really need it.
Creating a new GLib.Settings is not complicated too so there is no real need to share the same object everywhere in fact.